pkg/kubelet/cm/topologymanager/policy_single_numa_node.go

```diff
@@ -164,16 +175,25 @@ func (p *singleNumaNodePolicy) mergeProvidersHints(providersHints []map[string][
 		return defaultHint
 	}
 
-	allResourcesHints = p.filterHints(allResourcesHints)
-	// if no hints or there is a resource with empty hints after filtering, then policy cannot be satisfied
+	var noAffinityPreferredOnly bool
+	allResourcesHints, noAffinityPreferredOnly = p.filterHints(allResourcesHints)
+	// If no hints, then policy cannot be satisfied
 	if len(allResourcesHints) == 0 {
 		klog.Infof("[topologymanager] No hints that align to a single NUMA node.")
 		return defaultHint
 	}
+	// If there is a resource with empty hints after filtering, then policy cannot be satisfied.
+	// In the event that the only hints that exist are {nil true} update default hint preferred
+	// to allow scheduling.
 	for _, hints := range allResourcesHints {
 		if len(hints) == 0 {
 			klog.Infof("[topologymanager] No hints that align to a single NUMA node for resource.")
-			return defaultHint
+			if !noAffinityPreferredOnly {
+				return defaultHint
+			} else if noAffinityPreferredOnly {
+				defaultHint.Preferred = true
+				return defaultHint
+			}
```

Comment:
I think we should just get rid of the notion of a `defaultHint` in this function altogether. It confuses things, and it is sufficient to just return an appropriate hint at all points without it.

Reply:
Makes sense, I'll make these changes. My guess is the original idea was to use `defaultHint` with the `defaultAffinity` because when the first PR was created `canAdmitPodResult` used to do a check on `hint.NUMANodeAffinity.Count() != 1`. Whereas now it just checks `!hint.Preferred`, so the affinity is kind of irrelevant.